### ISSUE TITLE
How to connect through TCP/IP address?

### ISSUE THREAD

**Issue by Powherful** [Genymobile/scrcpy#1695]
I don't have the same issue that OP is having (I'm only ever connecting to two devices, my phone and my tablet), but +1 to this idea. It would just be a nice small time-saving feature to have, being able to type something like `scrcpy -t 1` instead of `scrcpy -s 192.168.1.6:5555`.

_Originally posted by @richbordoni in https://github.com/Genymobile/scrcpy/issues/1148#issuecomment-615497797_

I tried this method didn't workout for me.
Can you show some light towards this problem I am facing.

 MY OUTPUT:

Z:\andriod_to_pc_screensharing\scrcpy-win64-v1.16>scrcpy -s 192.168.225.58:5555 INFO: scrcpy 1.16 <https://github.com/Genymobile/scrcpy> adb: error: failed to get feature set: device '192.168.225.58:5555' not found ERROR: "adb push" returned with value 1 Press any key to continue...

**Comment by sdnazers20** [comment]
Hi 

I am able connect my mobile device using local network and able to mirror the screen, however I want to know if i can connect to my mobile using mobile network (3G/4G) by opening any port or so....
Please advice or provide a link to the solution

**Comment by rom1v** [comment]
> however I want to know if i can connect to my mobile using mobile network (3G/4G) by opening any port or so

No, you can't connect directly, because mobile networks typically do not allow to connect to your device from the internet (you can't "open a port on your NAT").

**Comment by sdnazers20** [comment]
> > however I want to know if i can connect to my mobile using mobile network (3G/4G) by opening any port or so
> 
> No, you can't connect directly, because mobile networks typically do not allow to connect to your device from the internet (you can't "open a port on your NAT").

Can you suggest me any other way to accomplish this basically i am trying yo mirror the screen over 3G/4G connection....your suggestion means a lot please advice

**Comment by igolf-plat** [comment]
> > > however I want to know if i can connect to my mobile using mobile network (3G/4G) by opening any port or so
> > 
> > 
> > No, you can't connect directly, because mobile networks typically do not allow to connect to your device from the internet (you can't "open a port on your NAT").
> 
> @rom1v Can you suggest me any other way to accomplish this basically i am trying yo mirror the screen over 3G/4G connection....your suggestion means a lot please advice

@sdnazers20 I'm doing the same thing. Basically you have to have a server with public IP to forward the data. And I've succeeded in connecting to ssh server running on the device behind nat, but when I change the to-be-forwarded port 22 to 5555, I fail in `
adb connect x.x.x.x:5555`

still struggle in reading adbd source code.

**Comment by fusionneur** [comment]
I created this tool to ease wireless connectivity. Check it out and let me know of your experience.

https://github.com/fusionneur/ethernally

**Comment by obenitez01** [comment]
> > > > sin embargo, quiero saber si puedo conectarme a mi móvil usando la red móvil (3G / 4G) abriendo cualquier puerto o algo así
> > > 
> > > 
> > > No, no puede conectarse directamente, porque las redes móviles normalmente no permiten conectarse a su dispositivo desde Internet (no puede "abrir un puerto en su NAT").
> > 
> > 
> > @ rom1v ¿Puedes sugerirme alguna otra forma de lograr esto? Básicamente estoy tratando de reflejar la pantalla a través de una conexión 3G / 4G ... tu sugerencia significa mucho por favor consejo
> 
> @ sdnazers20 Estoy haciendo lo mismo. Básicamente tienes que tener un servidor con IP pública para reenviar los datos. Y logré conectarme al servidor ssh que se ejecuta en el dispositivo detrás de nat, pero cuando cambio el puerto 22 a reenviar a 5555, fallo en` adb connect x.x.x.x:5555`
> 
> Todavía tengo problemas para leer el código fuente de ADBD.

Estoy buscando una forma de hacer lo mismo, @igolf-plat  aun no lo pudiste resolver ? si ya lo haz resuelto seria genial saber como lo hiciste.

**Comment by ncdejito** [comment]
> however I want to know if i can connect to my mobile using mobile network (3G/4G) by opening any port or so

was able to make it work with ngrok! 

![proof](https://user-images.githubusercontent.com/13001116/171670122-a545863c-5c02-415b-ab33-eda7080b1965.png)

Documented my steps [here](https://github.com/ncdejito/scrcpy-ngrok/blob/main/run.sh). Idea is to open a tunnel then run:
```
adb connect 123.tcp.ngrok.io:123456
scrcpy
```

**Comment by elasticdotventures** [comment]
the trick is wireless debugging, then go into pairing.
you can't connect directly - you must pair first.
